### ISSUE TITLE
StaticRoute cannot infer type for optional static_params and panics when provided a default

### ISSUE THREAD

**Issue by lukexor** [leptos-rs/leptos#1843]
**Describe the bug**
Attempting to use the new `<StaticRoute />` feature in 0.5.0

**Leptos Dependencies**

Please copy and paste the Leptos dependencies and features from your `Cargo.toml`.

For example:
```toml
leptos = { version = "0.5", features = ["rustls"] }
leptos_meta = { version = "0.5" }
leptos_dom = { version = "0.5" }
leptos_router = { version = "0.5" }
leptos_axum = { version = "0.5", optional = true }
```

**To Reproduce**

1. Create a new,  minimal leptos project with the above dependencies (e.g. `todo_app_sqlite_axum`)
2. Replace the existing routes line with the static generation lines in `main.rs`:
```rust
let (routes, static_data_map) = leptos_axum::generate_route_list_with_ssg(TodoApp);
leptos_router::build_static_routes(&leptos_options, TodoApp, &routes, &static_data_map).await.unwrap();
```
4. Change the `<Route />` component in `todo.rs` to `<StaticRoute />`:
```rust
<StaticRoute path="" view=Todos />
```
5. Get a type inference error:
```txt
error[E0282]: type annotations needed
   --> src/todo.rs:108:22
    |
108 |                     <StaticRoute path="" view=Todos/>
    |                      ^^^^^^^^^^^ cannot infer type of the type parameter `S` declared on the function `StaticRoute`
    |
help: consider specifying the generic arguments
    |
108 |                     <StaticRoute::<_, _, &str, S> path="" view=Todos/>
    |                                 +++++++++++++++++
```
7. Add the `static_params` prop:
```rust
<StaticRoute path="" view=Todos static_params=|| Box::pin(async { StaticParamsMap::default() }) />
```
9. Run `cargo leptos watch`
10. Receive a panic:
```txt
building static route:
thread 'main' panicked at /home/luke/dev/leptos/leptos_reactive/src/spawn.rs:87:13:
`spawn_local` called from outside of a `task::LocalSet`
stack backtrace:
   0: rust_begin_unwind
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/panicking.rs:597:5
   1: core::panicking::panic_fmt
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/core/src/panicking.rs:72:14
   2: tokio::task::local::spawn_local_inner
             at /home/luke/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/task/local.rs:347:21
   3: tokio::task::local::spawn_local
             at /home/luke/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.32.0/src/task/local.rs:337:9
   4: leptos_reactive::spawn::spawn_local
             at /home/luke/dev/leptos/leptos_reactive/src/spawn.rs:87:13
   5: leptos_reactive::resource::ResourceState<S,T>::load::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:1337:13
   6: leptos_reactive::memo::forward_ref_to::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/memo.rs:258:9
   7: leptos_reactive::signal::<impl leptos_reactive::node::NodeId>::try_with_no_subscription
             at /home/luke/dev/leptos/leptos_reactive/src/signal.rs:2041:12
   8: <leptos_reactive::memo::Memo<T> as leptos_reactive::signal::SignalWith>::try_with::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/memo.rs:472:13
   9: leptos_reactive::runtime::with_runtime::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:700:41
  10: std::thread::local::LocalKey<T>::try_with
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/thread/local.rs:270:16
  11: std::thread::local::LocalKey<T>::with
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/thread/local.rs:246:9
  12: leptos_reactive::runtime::with_runtime
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:695:13
  13: <leptos_reactive::memo::Memo<T> as leptos_reactive::signal::SignalWith>::try_with
             at /home/luke/dev/leptos/leptos_reactive/src/memo.rs:470:9
  14: leptos_reactive::resource::ResourceState<S,T>::load
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:1309:13
  15: leptos_reactive::resource::load_resource::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:391:13
  16: std::thread::local::LocalKey<T>::try_with
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/thread/local.rs:270:16
  17: std::thread::local::LocalKey<T>::with
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/thread/local.rs:246:9
  18: leptos_reactive::resource::load_resource
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:389:5
  19: leptos_reactive::resource::create_resource_helper::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:230:13
  20: <leptos_reactive::effect::EffectState<T,F> as leptos_reactive::effect::AnyComputation>::run
             at /home/luke/dev/leptos/leptos_reactive/src/effect.rs:366:25
  21: leptos_reactive::runtime::Runtime::update::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:230:25
  22: leptos_reactive::runtime::Runtime::with_observer
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:328:17
  23: leptos_reactive::runtime::Runtime::update
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:226:21
  24: leptos_reactive::runtime::Runtime::update_if_necessary
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:186:13
  25: leptos_reactive::effect::create_isomorphic_effect::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/effect.rs:256:9
  26: leptos_reactive::runtime::with_runtime::{{closure}}
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:700:41
  27: std::thread::local::LocalKey<T>::try_with
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/thread/local.rs:270:16
  28: std::thread::local::LocalKey<T>::with
             at /rustc/187b8131d4f760f856b214fce34534903276f2ef/library/std/src/thread/local.rs:246:9
  29: leptos_reactive::runtime::with_runtime
             at /home/luke/dev/leptos/leptos_reactive/src/runtime.rs:695:13
  30: leptos_reactive::effect::create_isomorphic_effect
             at /home/luke/dev/leptos/leptos_reactive/src/effect.rs:255:9
  31: leptos_reactive::resource::create_resource_helper
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:227:5
  32: leptos_reactive::resource::create_resource_with_initial_value
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:125:5
  33: leptos_reactive::resource::create_resource
             at /home/luke/dev/leptos/leptos_reactive/src/resource.rs:92:5
  34: todo_app_sqlite_axum::todo::Todos::__Todos
             at ./src/todo.rs:122:17
  35: todo_app_sqlite_axum::todo::Todos::{{closure}}
             at ./src/todo.rs:115:1
```

**Expected behavior**
I'd expect an optional `static_params` prop to actually be optional, and to not panic when provided.

Reference issues: #1594 and PR: #1649

**Comment by mrvillage** [comment]
Hey, I think the panic might be related to #1832, it seems to follow from the same set of circumstances (the traceback points to the part of the example containing resource and action code). As for the type var default, I've sunk way too much time into it and I haven't been able to find a way now or previously to get Rust to like it, as the leptos docs say Rust just doesn't like type var defaults when it comes to components. I don't like it either, but I can't think of any way to get it to behave nicely as intended. If you or anyone else has any ideas that'd work I'm all ears!

**Comment by gbj** [comment]
I think this is actually a more general issue and we'll need to provide an Axum-specific `build_static_routes` that runs it within a `LocalSet`, as the other response-generating functions in `leptos_axum` do, so that resources can be loaded successfuly. I played around with it a little this morning but had some errors and need to move onto other things.

@mrvillage When you built this for the PR were you using Actix for the server or Axum? I tested on Actix but realize I don't think I actually tested on Axum.
